src/functions-and-arrays.js

```diff
@@ -1,41 +1,162 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(a, b) {
+  if (a > b) {
+    return a;
+  } else {
+    return b;
+  }
+}
 
+maxOfTwoNumbers(8, 9);
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(longestWord) {
+  if (longestWord.length === 0) {
+    return null;
+  }
+  let magicWord = "";
+  let longest = 0;
+  longestWord.forEach(function(element) {
+    if(longest < element.length) {
+      longest = element.length;
+      magicWord = element
+    }
+  });
+  return magicWord;
+}
+
+findLongestWord(words);
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(num) {
 
 
 
+  if (!num) {
+    return 0;
+  }
+  let sum = 0;
+  num.forEach(function (element) {
+    sum += element;
+  })
+  return sum;
+}
+sumNumbers(numbers)
+console.log("3.0: " + sumNumbers(numbers))
 // Iteration #3.1 Bonus:
-function sum() {}
+const mezcladito = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+function sum(mixArr) {
+  
+  if(mixArr.length === 0) {
+    return 0;
+  } else {
+    let numArr = [];
+    let sumaNumArr = 0;
+    for(let y = 0; y < mixArr.length; y++) {
+      if(typeof mixArr[y] === 'boolean') {
+        if(mixArr[y]) {
+          numArr.push(1)
+        } else {
+          numArr.push(0)
+        }
+      } else if(typeof mixArr[y] === 'string') {
+        numArr.push(mixArr[y].length)
+      } else if (typeof mixArr[y] === 'number'){
+        numArr.push(mixArr[y])
+      }else{
+        return error('Invalid');
+      }
+    }
+    numArr.forEach(function(valor) {
+      sumaNumArr += valor;
+    })
+    
+    return sumaNumArr
```

Comment:
Las funciones nos ayudan a no reinventar la rueda te recomiendo hacer uso de las que ya creaste como en la linea de abajo 😉 
```suggestion
    return sumNumbers(numArr)
```